mmengine/structures/base_data_element.py

```diff
@@ -458,7 +460,14 @@ def set_field(self,
                 raise AttributeError(
                     f'Cannot set {name} to be a field of data '
                     f'because {name} is already a metainfo field')
-            self._data_fields.add(name)
+            # The name only added to `data_fields`` when it is not the
+            # attribute related to property(methods decorated by @property).
+            if not isinstance(
+                    getattr(
+                        type(self),
+                        inspect.getframeinfo(sys._getframe(1)).function, None),
+                    property):
```

Comment:
Why not directly use
```suggestion
            # The private field won't be recorded by `data_fields`.
            if not name.starts_with('_'):
```